db/note.php

```diff
@@ -31,21 +32,43 @@
  * @method void setContent(string $value)
  * @method boolean getFavorite()
  * @method void setFavorite(boolean $value)
+ * @method boolean getError()
+ * @method void setError(boolean $value)
+ * @method string getErrorMessage()
+ * @method void setErrorMessage(string $value)
  * @package OCA\Notes\Db
  */
 class Note extends Entity {
-
     public $etag;
     public $modified;
     public $title;
     public $category;
     public $content;
     public $favorite = false;
+    public $error = false;
+    public $errorMessage='';
 
     public function __construct() {
         $this->addType('modified', 'integer');
         $this->addType('favorite', 'boolean');
     }
+    /**
+     * @param File $file
+     * @return static
+     */
+    public static function fromFileMaybe(File $file, Folder $notesFolder, $tags=[]){
+
+        $id=$file->getId();
+
+        try{
+            $note=self::fromFile($file, $notesFolder, $tags=[]);
+        }catch(FileNotFoundException $e){
+            $note = self::fromException('File error: ('.$file->getName().') '.$e->getMessage(), $file, $notesFolder, array_key_exists($id, $tags) ? $tags[$id] : []);
+        }catch(DecryptionFailedException $e){
+            $note = self::fromException('Encryption Error: ('.$file->getName().')'.$e->getMessage(), $file, $notesFolder, array_key_exists($id, $tags) ? $tags[$id] : []);
+        }
```

Comment:
Can we add a `catch(\Exception)` here or merge all exceptions to a single catch here? Then we are prepared if a specific exception type is missing.